src/planning.py

```diff
@@ -823,7 +831,7 @@ class _DiscoveredFailure:
     failing_nsrt: _GroundNSRT
 
 
-class _DiscoveredFailureException(ExceptionWithInfo):
+class _DiscoveredFailureException(PlanningFailure):
```

Comment:
in order for this change to be okay, you need to make sure that wherever we're catching DiscoveredFailureException, we aren't already catching PlanningFailure inside of there, because now that inside catch would activate, changing the behavior of the code